### PR TITLE
DM-16405: add support for hue-preserving RGB

### PR DESCRIPTION
**Pull request by tgoldina**
https://jira.lsstcorp.org/browse/DM-16405

Added `set_stretch_hprgb`  to support hue-preserving rgb stretch and `band` argument to `set_stretch` to support 3-color per-band stretch. 

To test, use examples/basic_3color.ipynb

### CODE REVIEW

**Comment by tgoldina** [comment]
Currently docs build is failing in travis with the following exception:
```
Exception occurred:
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/sphinxcontrib/bibtex/directives.py", line 29, in <module>
    logger = sphinx.util.logging.getLogger(__name__)
AttributeError: module 'sphinx.util.logging' has no attribute 'getLogger'
```
`sphinxcontrib-bibtex` is sphinxcontrib_bibtex-0.4.1-py2.py3-none-any.whl was released Nov 28.

I am not sure what to do about it. @stargaser could you please advise?

**Comment by tgoldina** [comment]
@jonathansick still struggling with docs dependencies. Please advise.

**Comment by jonathansick** [comment]
@tgoldina I've had a chance to try the build, and we'll upgrade both documenteer and move from ltd-mason to ltd-conveyor since the old upload client has dependencies that don't seem to be compatible anymore. I'll send a PR shortly and instructions on how to set up Travis again. In the meantime, I suggest you remove the `pip install requirements.txt`, Sphinx `make html` and `ltd-mason` steps from the Travis file to get the build to pass.

**Comment by tgoldina** [comment]
@jonathansick Thank you for your suggestions in `dm-docs`. With the modified requirements.txt, the documentation build and travis checks seem to succeed. I can see the updated docs under `https://firefly-client.lsst.io/v/dm-16405_stretch_rgb/`. Do we still need to upgrade?

File: firefly_client/firefly_client.py [hunk 1]
```diff
@@ -1413,28 +1416,93 @@ def set_stretch(self, plot_id, stype=None, algorithm=None, **additional_params):
...
+        plot_id : `str` or `list` of `str`
+            ID of the plot to be stretched. If `plot_id` is a list or tuple, then each plot in the list
+            or the tuple is stretched in order.
+        asinh_q_value : `float`, optional
+            The asinh softening parameter for Asinh stretch.
+            Use Q=0 for linear stretch, increase Q to make brighter features visible.
+            When not specified, Q is calculated by Firefly to use full color range for intensity.
+        scaling_k : `float` or `list` of `float`, optional
+            Scaling coefficient from 0.1 to 10 (the default is 1).
+        pedestal_type : {'percent','minmax','absolute','zscale','sigma'} or `list` of `str`, optional
```
Comment (stargaser) [inline firefly_client/firefly_client.py]:
```suggestion
        pedestal_type : {'percent', 'minmax', 'absolute', 'zscale', 'sigma'} or `list` of `str`, optional
```

Spaces after the commas, will make the Sphinx output look better, I think.

File: firefly_client/firefly_client.py [hunk 2]
```diff
@@ -1413,28 +1416,93 @@ def set_stretch(self, plot_id, stype=None, algorithm=None, **additional_params):
...
         payload = {'stretchData': st_data, 'plotId': plot_id}
 
         return_val = self.dispatch_remote_action_by_post(self.channel, FireflyClient.ACTION_DICT['StretchImage'], payload)
         return_val['rv_string'] = serialized_rv
         return return_val
 
+    def set_stretch_hprgb(self, plot_id, asinh_q_value=None, scaling_k=1.0,
+                          pedestal_value=1, pedestal_type='percent'):
+        """
+        Change the stretch of RGB image (hue-preserving rgb case). When a parameter is list,
```
Comment (stargaser) [inline firefly_client/firefly_client.py]:
```suggestion
        Change the stretch of RGB image (hue-preserving rgb case). When a parameter is a list,
```

File: firefly_client/firefly_client.py [hunk 3]
```diff
@@ -1413,28 +1416,93 @@ def set_stretch(self, plot_id, stype=None, algorithm=None, **additional_params):
...
+        the first value if for red, the second value - for green, and the third value - for blue band.
```
Comment (stargaser) [inline firefly_client/firefly_client.py]:
```suggestion
        it must contain three elements, for red, green and blue bands respectively.
```

File: firefly_client/firefly_client.py [hunk 4]
```diff
@@ -1413,28 +1416,93 @@ def set_stretch(self, plot_id, stype=None, algorithm=None, **additional_params):
...
+        Otherwise the same value is used for all three bands.
```
Comment (stargaser) [inline firefly_client/firefly_client.py]:
```suggestion
        Otherwise the parameter is a scalar that is used for all three bands.
```

**Review by stargaser — APPROVED** [review]
Very nice work! Looks good to me. I have made some minor comments on the docstring for the new method.

**Pull request merged**